src/rules/selector-pseudo-class-whitelist/index.js

```diff
@@ -0,0 +1,58 @@
+import {
+  isStandardSyntaxSelector,
+  matchesStringOrRegExp,
+  parseSelector,
+  report,
+  ruleMessages,
+  validateOptions,
+} from "../../utils"
+import { isString } from "lodash"
+import { vendor } from "postcss"
+
+export const ruleName = "selector-pseudo-class-whitelist"
+
+export const messages = ruleMessages(ruleName, {
+  rejected: (selector) => `Unexpected pseudo-class "${selector}"`,
+})
+
+function rule(whitelist) {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, {
+      actual: whitelist,
+      possible: [isString],
+    })
+    if (!validOptions) { return }
+
+    root.walkRules(rule => {
+      const { selector } = rule
+
+      if (!isStandardSyntaxSelector(selector)) { return }
+      if (selector.indexOf(":") === -1) { return }
+
+      parseSelector(selector, result, rule, selectorTree => {
+        selectorTree.walkPseudos(pseudoNode => {
+          const pseudo = pseudoNode.value
+
+          // Ignore pseudo-elements
+          if (pseudo.indexOf("::") !== -1) { return }
```

Comment:
Actually, this is fine. The parser extracts and iterates through *singular* pseudos e.g.:

```js
:hover
:nth-child
:root
:HOVER
:hover
:nth-child
::before
:nth-child
::before
:-moz-placeholder
:-MOZ-PLACEholder
:focus
:nth-LAST-child
:global
:-ms-input-placeholder
:-Ms-INPUT-placeholder
:not
::selection
:nth-child
:nth-LAST-child
:hover
```

So, the colons will always be at the start. I've added a test for this.